### PR TITLE
Feature/setup pwa

### PR DESCRIPTION
**Pull request by DrewBradley**
## *Purpose:*
*What does this merge do? Check all that apply.*
- [x] Adds a new feature
- [x] Improves an existing feature
- [ ] Fixes a bug
- [ ] Cleans up (logs, tests, etc.)

## *Description: What does this PR do?*
Added service worker and updated manifest.json to convert application to progressive web application.
Added various links to allow for splash screen to display on iOS devices.
Added function to check online/offline status, and disable certain features while offline.
  - User cannot add event, contact, or agenda item while offline.
Added varying sizes of icons.

## *Instructions for reviewer:*
*Where should the reviewer start?*
`public/index.html`
`public/manifest.json`
`public/sw.js`
`src/components/agenda/AgendaSideBar.js`
`src/components/contacts/Contacts.js`
`src/components/form/Form.js`
`src/index.js`
`src/serviceWorkerRegistration.js`

*How should this be manually tested?*
Open application locally.
Open Chrome devtools and toggle to mobile view.
At the top of the screen where it says "No throttling" click and select "Offline"
Application should still work as normal.
Try to add an event, contact, or agenda item. An error should display.
*Any background context you want to provide?*
Screen Shot 2021-04-20 at 12 45 20 PM
Screen Shot 2021-04-20 at 12 45 02 PM

*What are the relevant tickets?*

## *Issues:*
*Are there any concerns, issues, or bugs in this branch? If so describe them.*
When refreshing the app from the service worker, occasionally an error occurs.
To unregister the service worker and refresh the app, press and hold 'shift' and click the refresh icon on the browser.

## Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have used tests to prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my change

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -2,16 +2,26 @@
...
+    <link href="icon-256x256.png" media="(device-width: 375px) and (device-height: 812px) and (-webkit-device-pixel-ratio: 3)" rel="apple-touch-startup-image" />
+    <link href="icon-384x384.png" media="(device-width: 414px) and (device-height: 896px) and (-webkit-device-pixel-ratio: 2)" rel="apple-touch-startup-image" />
+    <link href="icon-384x384.png" media="(device-width: 414px) and (device-height: 896px) and (-webkit-device-pixel-ratio: 3)" rel="apple-touch-startup-image" />
+    <link href="icon-384x384.png" media="(device-width: 768px) and (device-height: 1024px) and (-webkit-device-pixel-ratio: 2)" rel="apple-touch-startup-image" />
+    <link href="icon-512x512.png" media="(device-width: 834px) and (device-height: 1112px) and (-webkit-device-pixel-ratio: 2)" rel="apple-touch-startup-image" />
+    <link href="icon-512x512.png" media="(device-width: 834px) and (device-height: 1194px) and (-webkit-device-pixel-ratio: 2)" rel="apple-touch-startup-image" />
+    <link href="icon-512x512.png" media="(device-width: 1024px) and (device-height: 1366px) and (-webkit-device-pixel-ratio: 2)" rel="apple-touch-startup-image" />
```
Comment (kristenmb) [inline public/index.html:24]: so many icons wow

File: src/components/agenda/AgendaSideBar.js [hunk 2]
```diff
@@ -9,6 +9,10 @@ function AgendaSideBar({currentEvent}) {
 
   const onSubmit = (e) => {
     e.preventDefault()
+    if (!navigator.onLine) {
```
Comment (kristenmb) [inline src/components/agenda/AgendaSideBar.js:12]: This was so clever. I love how this works!

**Review by kristenmb — APPROVED** [review]

File: public/index.html (hunk 1 above)
Comment (richardltyler) [reply public/index.html:24]: This is a lot of work. Icon't believe you've done this.

File: src/components/contacts/Contacts.js [hunk 3]
```diff
@@ -55,6 +57,13 @@ const Contacts = () => {
 
   const toggleAddContactModule = (event) => {
     event.preventDefault();
+    if (!navigator.onLine) {
```
Comment (richardltyler) [inline src/components/contacts/Contacts.js:60]: Good UX detail

**Review by richardltyler — APPROVED** [review]
Good stuff! I can't wait to download this

**Pull request merged**